### PR TITLE
hll optimize: trace memory usage, new explicit data when really need

### PR DESCRIPTION
**Pull request by zuochunwei**
reduce hll memory occupied：

replace uint64_t _explicit_data[160*2] with uint64_t*
new memory for explicit data when really needed
trace HLL memory usage

### CODE REVIEW

**Comment by morningman** [comment]
Hi @zuochunwei Could you please create an issue to describe more detail about how this works and how much memory can be reduced after this PR?

**Comment by zuochunwei** [comment]
> Hi @zuochunwei Could you please create an issue to describe more detail about how this works and how much memory can be reduced after this PR?

each HLL object will reduce nearly 2k bytes，because 160*2*sizeof(uint64_t) result 2560，almost 2k bytes.
so when use tool to import data，if the table have hundreds of millions records,  we generate one hll object for each record,  the memory usage will increase hundreds Gigabytes

**Review by yangzhg — DISMISSED** [review]
LGTM

File: be/src/olap/hll.h [hunk 1]
```diff
@@ -33,6 +33,7 @@ class Slice;
 const static int HLL_COLUMN_PRECISION = 14;
 const static int HLL_ZERO_COUNT_BITS = (64 - HLL_COLUMN_PRECISION);
 const static int HLL_EXPLICIT_INT64_NUM = 160;
+const static int HLL_EXPLICIT_INT64_NUM_DOUBLE = HLL_EXPLICIT_INT64_NUM*2;
```
Comment (HappenLee) [inline be/src/olap/hll.h:36]: HLL_EXPLICIT_INT64_NUM * 2

File: be/src/olap/hll.h [hunk 2]
```diff
@@ -154,7 +174,7 @@ class HyperLogLog {
     HllDataType _type = HLL_DATA_EMPTY;
 
     uint32_t _explicit_data_num = 0;
-    uint64_t _explicit_data[HLL_EXPLICIT_INT64_NUM * 2];
+    uint64_t* _explicit_data = nullptr;
```
Comment (HappenLee) [inline be/src/olap/hll.h:177]: directly use the raw ptr is dangerous and complex. Rethink shoule we replace is use std::unique_ptr

Comment (zuochunwei) [reply be/src/olap/hll.h:177]: no need replace with unique_ptr,  keep style consistency with uint8_t *_registers

File: be/src/olap/hll.h (hunk 1 above)
Comment (zuochunwei) [reply be/src/olap/hll.h]: fixed it

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**